### PR TITLE
fix memory leak when download 300+ files

### PR DESCRIPTION
**Pull request by shcheuk**
fix issue https://github.com/expo/expo/issues/4239

# Why

I was forced to update my  released app due to an outdated facebook SDK , expo SDK from 28 to 32.
The update cause all android user crash when installing the app, the installation process include downloading 7000+ files. Digging in problem, heap size keep going up when download and app was crashed by out of memory or max heap size exceeded.It does not happen on SDK 28 nor SDK 32 iOS. Issue is confined to android.

# How

By following the doc from okhttp here:
https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.html and there:
https://stackoverflow.com/questions/48532860/is-it-thread-safe-to-make-calls-to-okhttpclient-in-parallel

basically just make the client share (singleton).

# Test Plan

To test this change use the repo in https://github.com/expo/expo/issues/4239, see if the 10000 download test succeed. (hopefully can make this test in the circle ci, as multi files download job easily break from SDK to SDK. If someone do, please change the sample download files as those are from our amazon cloudfront, my boss will ask me pay the bill  )

### CODE REVIEW

**Comment by shcheuk** [comment]
Oh I am sorry, I just read that:
Expo client
Please check with us before putting work into a Pull Request! We don't yet have a good guide available that covers the nuances of how to work with the Expo client so you will want a direct line of communication with someone on the team to ask us questions. The best place to talk to us is either on Slack at https://slack.expo.io or the forums at https://forums.expo.io.
will go back to forum.expo.io for this....
